ir/type.cpp

```diff
@@ -87,41 +88,65 @@ const Type_Bits *Type_Bits::get(int width, bool isSigned) {
     return result;
 }
 
-const Type::Unknown *Type::Unknown::get() {
-    static const Type::Unknown *singleton = nullptr;
-    if (!singleton) singleton = (new Type::Unknown());
+const Type_Bits *Type_Bits::get(const Util::SourceInfo &si, int sz, bool isSigned) {
+    auto *result = new IR::Type_Bits(si, sz, isSigned);
+    if (sz < 0) {
+        ::error(ErrorType::ERR_INVALID, "%1%: Width of type cannot be negative", result);
```

Comment:
Should this be `BUG_CHECK` instead? Also, it seems we're creating `result` here just for some message printing. I think we can just use `sz` and create result only when all sanity checks pass

Reply:
I am not sure why an error was chosen here but the comment 
```
// Return a value that will not cause crashes later on
```
indicates this was deliberate? 

I moved the result allocation down and changed the wording a bit.